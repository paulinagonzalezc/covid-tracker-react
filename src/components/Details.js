import React, { useEffect } from 'react';
import { useParams, NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import { fetchDetail } from '../redux/details';

const Details = () => {
  const routeParams = useParams();
  const detail = useSelector((state) => state.details);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDetail(routeParams.name));
  }, [dispatch]);
  return (
    <div className="details-container">
      <NavLink to="/"><ArrowBackIcon /></NavLink>
      <h1>{detail.country}</h1>
      <img className="detail-image" alt="flag" src={detail.flag} />
      <ul className="country-details">
        <li>
          Today Cases:
          {detail.todayCases}
        </li>
        <li>
          Total Cases:
          {detail.cases}
        </li>
        <li>
          Today Deaths:
          {detail.todayDeaths}
        </li>
        <li>
          Total Deaths:
          {detail.deaths}
        </li>
        <li>
          Country Population:
          {detail.population}
        </li>
      </ul>
    </div>
  );
};

export default Details;
