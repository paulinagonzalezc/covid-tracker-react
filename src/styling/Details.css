.country-name {
  color: white;
  font-weight: 300;
  margin: 0;
  width: 60%;
  padding: 10px 0;
  text-align: center;
  background-color: #35548b;
  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
}

.details-container {
  margin-top: 30px;
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: 1fr 2fr 3fr;
}

.detail-image {
  width: 250px;
  justify-self: center;
}

.arrow-back {
  color: white;
  justify-self: end;
}

.detail {
  list-style-type: none;
  color: white;
  display: flex;
  justify-content: space-between;
  padding: 15px;
  margin-top: 15px;
  background-color: #4269b2;
  box-shadow: rgba(0, 0, 0, 0.24) 0 3px 8px;
}

.country-details {
  padding: 0;
  margin: 0;
  width: 60%;
}

@media only screen and (max-width: 600px) {
  .details-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .country-details {
    padding: 0;
    margin: 0;
    width: 100%;
  }

  .country-name {
    width: 100%;
  }

  .detail-image {
    justify-self: center;
    margin: 20px 0;
  }
}
