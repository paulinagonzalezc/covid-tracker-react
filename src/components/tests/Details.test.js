import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Details from '../Details';

describe('Details component test', () => {
  test('should render', () => {
    const details = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
        ,
      </Provider>,
    );
    expect(details).toMatchSnapshot();
    const title = screen.getByText(/Today Cases:/);
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('today-cases');
  });
});