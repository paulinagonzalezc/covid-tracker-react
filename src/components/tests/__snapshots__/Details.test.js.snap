// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-container"
      >
        <a
          aria-current="page"
          class="arrow-back active"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </a>
        <img
          alt="flag"
          class="detail-image"
        />
        <div
          class="country-wrapper"
        >
          <h1
            class="country-name"
          />
          <ul
            class="country-details"
            data-testid="country-details"
          >
            <li
              class="detail"
            >
              <span
                class="today-cases"
              >
                Today Cases:
              </span>
              <span />
            </li>
            <li
              class="detail"
            >
              <span
                class="total-cases"
              >
                Total Cases:
              </span>
              <span />
            </li>
            <li
              class="detail"
            >
              <span
                class="today-deaths"
              >
                Today Deaths:
              </span>
              <span />
            </li>
            <li
              class="detail"
            >
              <span
                class="total-deaths"
              >
                Total Deaths:
              </span>
              <span />
            </li>
            <li
              class="detail"
            >
              <span>
                Country Population:
              </span>
              <span />
            </li>
          </ul>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="details-container"
    >
      <a
        aria-current="page"
        class="arrow-back active"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </a>
      <img
        alt="flag"
        class="detail-image"
      />
      <div
        class="country-wrapper"
      >
        <h1
          class="country-name"
        />
        <ul
          class="country-details"
          data-testid="country-details"
        >
          <li
            class="detail"
          >
            <span
              class="today-cases"
            >
              Today Cases:
            </span>
            <span />
          </li>
          <li
            class="detail"
          >
            <span
              class="total-cases"
            >
              Total Cases:
            </span>
            <span />
          </li>
          <li
            class="detail"
          >
            <span
              class="today-deaths"
            >
              Today Deaths:
            </span>
            <span />
          </li>
          <li
            class="detail"
          >
            <span
              class="total-deaths"
            >
              Total Deaths:
            </span>
            <span />
          </li>
          <li
            class="detail"
          >
            <span>
              Country Population:
            </span>
            <span />
          </li>
        </ul>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
